main.tf

```diff
@@ -442,7 +442,7 @@ resource "aws_db_parameter_group" "this" {
 ################################################################################
 
 resource "aws_cloudwatch_log_group" "this" {
-  for_each = toset([for log in var.enabled_cloudwatch_logs_exports : log if local.create_cluster && var.create_cloudwatch_log_group])
+  for_each = toset([for log in var.enabled_cloudwatch_logs_exports : log if local.create_cluster && var.create_cloudwatch_log_group && !var.cluster_use_name_prefix])
```

Comment:
Updated cloudwatch log group creation based on [this feedback](https://github.com/terraform-aws-modules/terraform-aws-rds/issues/477) which is also relevant to this module.